### DISCUSSION TITLE
"Inset" lonely window on ultrawide monitor? Aka smaller size/large gaps ONLY when there's One Window active (per workspace)

### DISCUSSION THREAD

**Discussion by fZ0x539** [i3/i3#5445] [Q&A]
Hopefully I described it well enough in the title. 
I'm looking for a way to make it so that the first window inside a workspace is 'inset' and smaller sized than it usually would. I'm using a 3440x1440 34" monitor so it stretches far too wide, and I don't want to $mod+space then have to re-do it once I open a second one.. It gets too messy.

I've looked around for alternatives and heard of [con_mark="lonely"], but there's apparently no way for it to mark the first window as lonely and then unmark it the moment I open another one, you have to manually toggle lonely on/off which is just as tedious as floating.

**Comment by okraits** [comment]
You can write a script using the IPC interface which looks for workspace events: https://i3wm.org/docs/ipc.html#_workspace_event

* Watch for workspace events (change: init, I guess)
* Modify the first window in the way you want
